core/authentication_api.php

```diff
@@ -209,7 +209,7 @@ function auth_auto_create_user( $p_username, $p_password ) {
 
 	if( $t_auto_create ) {
 		# attempt to create the user
-		$t_cookie_string = user_create( $p_username, md5( $p_password ) );
+		$t_cookie_string = user_create( $p_username, "" );
```

Comment:
We already update the user often, e.g. last visit which is updated on every page, so I wouldn't over think it.  There is really two options to clear existing data:
- Generate a random password.
- Add a schema update step to reset all passwords in case of LDAP -- I think this is risky since if someone has an approach where they having mixed login, we will delete all their user passwords.

Now if you want ! + random hash, so that you can figure it out and avoid an extra update, then I'm OK with that.  Just make sure the final string fits.  Also you may want a function that generates this string and is used from the appropriate locations.

For example:
- ldap_generate_random_password()
- ldap_is_random_password()